src/occupation.jl

```diff
@@ -9,35 +21,31 @@ function find_occupation_gap_zero_temperature(basis, energies, Psi)
     n_bands = size(Psi[1], 2)
     n_electrons = basis.model.n_electrons
     T = eltype(basis.kpoints[1].coordinate)
-
-    @assert basis.model.spin_polarisation in (:none, :spinless)
-    if basis.model.spin_polarisation == :none
-        @assert n_electrons % 2 == 0
-        n_occ = div(n_electrons, 2)
-        filled_occ = 2
-    else
-        n_occ = n_electrons
-        filled_occ = 1
-    end
-    @assert n_bands ≥ n_occ
     @assert basis.model.temperature == 0.0
 
+    filled_occ = filled_occupation(basis)
+    n_fill = div(n_electrons, 2)
```

Comment:
This is not true for `spin_polarisation == :collinear` for example. I'd suggest to add a comment and an assert.